src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -13288,7 +13288,7 @@ Real64 VRFCondenserEquipment::VRFOU_FlowRate(EnergyPlusData &state,
         deltaT = this->C3Tc * pow_2(SHSC) + this->C2Tc * SHSC + this->C1Tc;
         T_coil_surf = TeTc - deltaT;
         T_coil_out = T_coil_in + (T_coil_surf - T_coil_in) * (1 - BF);
-        m_air = Q_coil / (T_coil_out - T_coil_in) / 1005.0;
+        m_air = abs(Q_coil) / (T_coil_out - T_coil_in) / 1005.0;
```

Comment:
If I'm understanding `Q_coil` correctly, it should never be <0. While this fixes the issue (which occurs only a few timesteps in the year) it masks a problem higher up. The Q that is passed in to this function shouldn't be <0, ever.

Reply:
So Q_coil is negative when it's in cooling mode. It is positive when it's heating

Reply:
> So Q_coil is negative when it's in cooling mode. It is positive when it's heating

I'm not sure that's the case.  `Q_coil` here is the heat rejection when the unit is in cooling mode. So it's heating the outdoor air, so (Tcoilout-Tcoilin) should be positive, so you would expect `Q_coil` to be positive.

In the next block where it's in heating mode (so it's cooling the outdoor air) delta is `(h_coil_in - h_coil_out` so, again deltaH is positive and so `Q_coil` should be positive.

I ran in the debugger and saw both positive and negative values of `Q_coil` in the `OperationMode == HXOpMode::CondMode` block. Tried to watch the values in the `OperationMode == HXOpMode::EvapMode` block, but it never goes there. Even with Atlanta weather. That doesn't seem correct. Is it possible that the value passed in for `OperationMode` is not correct here?

Reply:
Now I'm not certain whether `Q_coil` should be always positive or not.
Although, from the definition of this function, it seems they do need to be absolute value of `Q_coil`.

```
Real64 VRFCondenserEquipment::VRFOU_FlowRate(EnergyPlusData &state,
                                             HXOpMode const OperationMode, // Mode 0 for running as condenser, 1 for evaporator
                                             Real64 const TeTc,            // VRF Tc at cooling mode, or Te at heating mode [C]
                                             Real64 const SHSC,            // SC for OU condenser or SH for OU evaporator [C]
                                             Real64 const Q_coil,          // absolute value of OU coil heat release or heat extract [W] <----- here they require absolute value
                                             Real64 const T_coil_in,       // Temperature of air at OU coil inlet [C]
                                             Real64 const W_coil_in        // Humidity ratio of air at OU coil inlet [kg/kg]
) const
{
```

Maybe I should just add a `Q_coil = abs(Q_coil)` at the beginning of the call to this function? @mjwitte

Reply:
> Maybe I should just add a Q_coil = abs(Q_coil) at the beginning of the call to this function?

No, I think that will mask other problems. To me, the "absolute value" comment implies that the value passed in will always be positive. We need to understand if 'HXOpMode` is being passed in correctly and better understand the variables that are being passed in for `Q_coil`.

Reply:
I see. I will look into it. Currently there are definitely both positive and negatives for the same mode

Reply:
I've done some investigations and had a few findings. Although I'm not sure if this is enough to figure out a fix. @mjwitte 

## about the operation mode

According to the [documentation](https://bigladdersoftware.com/epx/docs/24-1/engineering-reference/variable-refrigerant-flow-heat-pumps.html#VRF-HR-operational-modes), there are 6 heat recovery modes
- Mode 1: Cooling load only. No heating load. Both OU heat exchangers operate as condensers.
- Mode 2: Simultaneous heating and cooling. The sum of the zone cooling loads and compressor heat is much larger than the sum of the zone heating loads. Both OU heat exchangers operate as condensers.
- Mode 3: Simultaneous heating and cooling. The sum of the zone cooling loads and compressor heat is slightly larger than the sum of the zone heating loads. One OU heat exchanger operates as a condenser while the other as an evaporator.
- Mode 4: Simultaneous heating and cooling. The sum of the zone cooling loads and compressor heat is slightly smaller than the sum of the zone heating loads. One OU heat exchanger operates as a condenser while the other as an evaporator.
- Mode 5: Simultaneous heating and cooling. The sum of the zone cooling loads and compressor heat is much smaller than the sum of the zone heating loads. Both OU heat exchangers operate as evaporators.
- Mode 6: Heating load only. No cooling load. Both OU heat exchangers operate as evaporators.

Here is how the heat recovery mode is determined
```

// Determine FlagToLower
if (state.dataEnvrn->OutDryBulbTemp - this->DiffOUTeTo < Tsuction) {
    temp_Tsuction = state.dataEnvrn->OutDryBulbTemp - this->DiffOUTeTo;
    FlagToLower = true;
} else {
    temp_Tsuction = Tsuction;
    FlagToLower = false;
}

// Calculate compressor speed satisfying IU loads: rps1_evap & rps2_cond
this->VRFOU_CompSpd(state, Q_c_TU_PL, HXOpMode::EvapMode, temp_Tsuction, Tdischarge, h_IU_evap_in, h_IU_PLc_out, rps1_evap);
this->VRFOU_CompSpd(state, Q_h_TU_PL, HXOpMode::CondMode, temp_Tsuction, Tdischarge, h_IU_evap_in, h_IU_PLc_out, rps2_cond);

// Determine FlagMode5
if (rps1_evap <= rps2_cond) {
    FlagMode5 = true;
} else {
    FlagMode5 = false;
}

// Determine HR Mode
if (FlagMode5) {
    HRMode = 5;
    if (FlagToLower)
        HRMode_sub = 1;
    else
        HRMode_sub = 2;
} else {

    if (FlagToLower)
        HRMode = 3; // Mode 3&4 share the same logics below
    else
        HRMode = 2;
}
```

One issue I saw is that `rps2_cond` is negative sometimes. Here's some debugging print. I used the 
`VariableRefrigerantFlow_FluidTCtrl_HR_5Zone.idf` and `USA_FL_Miami.Intl.AP.722020_TMY3.epw` combination. For the 01/02 07, 4th time step, `rps2_cond` is negative. This is caused by 

```
01/02 07-4 HRMode=2, flag5=false, rps1_evap=14.796895201613392,rps2_cond=-103.04599155624868: Q_c_TU_PL=279.7809733457191, Q_h_TU_PL=300.59323456715885, temp_Tsuction=14.000000000000002,Tdischarge=36
```
This is caused by the following in `VRFCondenserEquipment::VRFOU_CompSpd`, where `Q_req` is smaller than the `CompEvaporatingPWRSpd(CounterCompSpdTemp)` when `CounterCompSpdTemp = 1`.
```
Q_cond_req = Q_req;
...
Q_evap_req = Q_cond_req - CompEvaporatingPWRSpd(CounterCompSpdTemp);
```

I was considering, maybe `Q_evap_req = Q_cond_req - CompEvaporatingPWRSpd(CounterCompSpdTemp);` should be `Q_evap_req = max(0.0, Q_cond_req - CompEvaporatingPWRSpd(CounterCompSpdTemp));`

Even with this, there still remains a question of how this "much larger" or "much smaller" in mode 2 and 5 is reflected in the code. I feel the current code seems to be just testing for "larger" or "smaller".

## about the `Q_coil` sign

the input argument `Q_h_OU` in `m_air_cond = this->VRFOU_FlowRate(
            state, HXOpMode::CondMode, Tdischarge, this->SC, Q_h_OU, state.dataEnvrn->OutDryBulbTemp, state.dataEnvrn->OutHumRat);` corresponds to the `Q_coil` referred to in the last comment.

Here is how it's computed. As `Q_c_tot + Ncomp` is smaller than `Q_h_TU_PL`, the `Q_h_OU` is negative.
```
// compressor: Ncomp & Q_c_tot
this->VRFOU_CompCap(state, CompSpdActual, Tsuction, Tdischarge, h_IU_evap_in, h_comp_in, Q_c_tot, Ncomp);

Real64 Q_h_tot;                 // Total condenser capacity [W]
Real64 Q_h_TU_PL;                // Heating load to be met at heating mode, including the piping loss (W)
Real64 &Q_h_OU,        // OU condenser load [W]
Q_h_tot = Q_c_tot + Ncomp; 
Q_h_OU = Q_h_tot - Q_h_TU_PL;
```
Following is the debug print
```
01/02 07-4 Q_h_OU=-14.889397577117904=285.70383699004094 - 300.59323456715885
```
For this specific instance, the `Q_h_TU_PL` is mostly on the piping loss term.
`Q_h_TU_PL = TU_HeatingLoad + Pipe_Q_h;`
```
01/02 07-4 Q_h_TU_PL=300.59323456715885=0.8205014721511905 + 299.77273309500765
```

Maybe being in HR mode 2 should have ensured `Q_c_tot + Ncomp` much larger than `Q_h_TU_PL`? But currently it's not.

Reply:
So during these iterations h_comp_out_new did not change but h_comp_out did change? That makes no sense. It feels like this iteration loop isn't updating something that SHOULD change in the h_comp_out_new equation.

Reply:
Yeah, for the specific time step when the fan power became negative, this is what happens. Maybe that could be another criteria (checking whether h_comp_out_new changes) to add to the convergence check